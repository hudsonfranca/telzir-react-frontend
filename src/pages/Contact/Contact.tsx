import React from 'react';

import { Parallax } from 'react-parallax';
import {
  Container,
  SectionOne,
  SectionTwo,
  CardTitle,
  Card,
  CardBody,
} from './styles';
import ContactImg from '../../assets/contact.jpg';
import question from '../../assets/question.jpg';
import call from '../../assets/call.jpg';
import sms from '../../assets/sms.jpg';
import suport from '../../assets/suport2.jpg';

const Contact: React.FC = () => {
  document.title = 'Contato';
  return (
    <Container data-test="contactContainer">
      <Parallax
        bgImage={ContactImg}
        strength={500}
        bgClassName="react-parallax-bgimage"
      >
        <SectionOne data-test="contactSectionOne">
          <h1>Fale com a Telzir e tire suas dúvidas</h1>

          <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi
            exercitationem provident facilis nulla nam vero quod est magnam
            dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad
            molestiae laboriosam quam?
          </p>
        </SectionOne>
      </Parallax>
      <SectionTwo data-test="contactSectionTwo">
        <Card bgUrl={sms} data-test="contactCard">
          <CardTitle>WHATSAPP</CardTitle>
          <CardBody>Seja atendido via WhatsApp</CardBody>
        </Card>
        <Card bgUrl={call} data-test="contactCard">
          <CardTitle>TELEFONE</CardTitle>
          <CardBody>Ligue grátis para a Telzir</CardBody>
        </Card>
        <Card bgUrl={question} data-test="contactCard">
          <CardTitle>OUVIDORIA</CardTitle>
          <CardBody>Sugestões, críticas e reclamações</CardBody>
        </Card>
        <Card bgUrl={suport} data-test="contactCard">
          <CardTitle>ENVIE SUA DÚVIDA</CardTitle>
          <CardBody>Informe o plano e o que precisa</CardBody>
        </Card>
      </SectionTwo>
    </Container>
  );
};

export default Contact;
