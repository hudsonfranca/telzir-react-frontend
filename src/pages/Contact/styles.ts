import styled from 'styled-components';

interface CardProps {
  bgUrl: string;
}

export const Container = styled.div`
  width: 100%;
  min-height: 100vh;
`;

export const SectionOne = styled.section`
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: var(--white);

  h1 {
    font-size: 4rem;
    padding-bottom: 20px;
    width: 80%;
    text-align: center;
  }

  p {
    font-size: 2rem;
    width: 80%;
    text-align: center;
  }
  &::after {
    content: '';
    width: 100%;
    height: 100%;
    background: rgba(11, 12, 16, 0.5);
    position: absolute;
    top: 0;

    z-index: -1;
  }
`;

export const SectionTwo = styled.section`
  width: 100%;
  min-height: 100vh;
  background-color: var(--secondary);
  padding: 20px;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-template-rows: repeat(auto-fit, minmax(250px, 1fr));
  grid-gap: 30px;
  justify-content: center;
  align-items: center;
`;

export const Card = styled.div<CardProps>`
  width: 100%;
  height: 100%;
  background-image: url(${({ bgUrl }) => bgUrl});

  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: flex;
  flex-direction: column;
  justify-content: start;
  align-items: flex-start;
  cursor: pointer;

  position: relative;
  z-index: 1;

  &::after {
    content: '';
    width: 100%;
    height: 100%;

    background-image: linear-gradient(
      100deg,
      var(--primary),
      rgba(31, 40, 51, 0.1)
    );

    position: absolute;
    top: 0;
    z-index: -1;
  }
`;

export const CardTitle = styled.p`
  font-size: 3rem;
  padding: 20px;
`;

export const CardBody = styled.p`
  font-size: 2rem;
  padding: 20px;
`;
