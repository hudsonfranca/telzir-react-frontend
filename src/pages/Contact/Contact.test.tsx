import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import Contact from './Contact';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<Contact />);
});

describe('<Home /> redering', () => {
  it('Should render without errors', () => {
    const home = renderer.create(<Contact />).toJSON();

    expect(home).toMatchSnapshot();
  });

  it('Should render 1 contactContainer', () => {
    expect(wrapper.find(`[data-test='contactContainer']`).length).toEqual(1);
  });

  it('Should render 1 contactSectionOne', () => {
    expect(wrapper.find(`[data-test='contactSectionOne']`).length).toEqual(1);
  });

  it('Should render 1 contactSectionTwo', () => {
    expect(wrapper.find(`[data-test='contactSectionTwo']`).length).toEqual(1);
  });

  it('Should render 4 contactCard', () => {
    expect(wrapper.find(`[data-test='contactCard']`).length).toEqual(4);
  });
});
