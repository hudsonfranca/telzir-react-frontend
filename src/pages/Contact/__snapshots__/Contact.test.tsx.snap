// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> redering Should render without errors 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
}

.c1 {
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--white);
}

.c1 h1 {
  font-size: 4rem;
  padding-bottom: 20px;
  width: 80%;
  text-align: center;
}

.c1 p {
  font-size: 2rem;
  width: 80%;
  text-align: center;
}

.c1::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.5);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c2 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--secondary);
  padding: 20px;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(250px,1fr));
  grid-template-rows: repeat(auto-fit,minmax(250px,1fr));
  grid-gap: 30px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 100%;
  height: 100%;
  background-image: url(sms.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  cursor: pointer;
  position: relative;
  z-index: 1;
}

.c3::after {
  content: '';
  width: 100%;
  height: 100%;
  background-image: linear-gradient( 100deg, var(--primary), rgba(31,40,51,0.1) );
  position: absolute;
  top: 0;
  z-index: -1;
}

.c6 {
  width: 100%;
  height: 100%;
  background-image: url(call.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  cursor: pointer;
  position: relative;
  z-index: 1;
}

.c6::after {
  content: '';
  width: 100%;
  height: 100%;
  background-image: linear-gradient( 100deg, var(--primary), rgba(31,40,51,0.1) );
  position: absolute;
  top: 0;
  z-index: -1;
}

.c7 {
  width: 100%;
  height: 100%;
  background-image: url(question.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  cursor: pointer;
  position: relative;
  z-index: 1;
}

.c7::after {
  content: '';
  width: 100%;
  height: 100%;
  background-image: linear-gradient( 100deg, var(--primary), rgba(31,40,51,0.1) );
  position: absolute;
  top: 0;
  z-index: -1;
}

.c8 {
  width: 100%;
  height: 100%;
  background-image: url(suport2.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  cursor: pointer;
  position: relative;
  z-index: 1;
}

.c8::after {
  content: '';
  width: 100%;
  height: 100%;
  background-image: linear-gradient( 100deg, var(--primary), rgba(31,40,51,0.1) );
  position: absolute;
  top: 0;
  z-index: -1;
}

.c4 {
  font-size: 3rem;
  padding: 20px;
}

.c5 {
  font-size: 2rem;
  padding: 20px;
}

<div
  className="c0"
  data-test="contactContainer"
>
  <div
    className="react-parallax "
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <img
      alt=""
      className="react-parallax-bgimage"
      src="contact.jpg"
      style={
        Object {
          "MozBackfaceVisibility": "hidden",
          "MsBackfaceVisibility": "hidden",
          "WebkitBackfaceVisibility": "hidden",
          "WebkitTransform": "translate3d(-50%, 0, 0)",
          "WebkitTransformStyle": "preserve-3d",
          "left": "50%",
          "position": "absolute",
          "transform": "translate3d(-50%, 0, 0)",
        }
      }
    />
    <div
      className="react-parallax-content"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <section
        className="c1"
        data-test="contactSectionOne"
      >
        <h1>
          Fale com a Telzir e tire suas dúvidas
        </h1>
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi exercitationem provident facilis nulla nam vero quod est magnam dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad molestiae laboriosam quam?
        </p>
      </section>
    </div>
  </div>
  <section
    className="c2"
    data-test="contactSectionTwo"
  >
    <div
      className="c3"
      data-test="contactCard"
    >
      <p
        className="c4"
      >
        WHATSAPP
      </p>
      <p
        className="c5"
      >
        Seja atendido via WhatsApp
      </p>
    </div>
    <div
      className="c6"
      data-test="contactCard"
    >
      <p
        className="c4"
      >
        TELEFONE
      </p>
      <p
        className="c5"
      >
        Ligue grátis para a Telzir
      </p>
    </div>
    <div
      className="c7"
      data-test="contactCard"
    >
      <p
        className="c4"
      >
        OUVIDORIA
      </p>
      <p
        className="c5"
      >
        Sugestões, críticas e reclamações
      </p>
    </div>
    <div
      className="c8"
      data-test="contactCard"
    >
      <p
        className="c4"
      >
        ENVIE SUA DÚVIDA
      </p>
      <p
        className="c5"
      >
        Informe o plano e o que precisa
      </p>
    </div>
  </section>
</div>
`;
